packages/react-composites/src/composites/ChatComposite/FileDownloadCards.tsx

```diff
@@ -48,20 +52,34 @@ export const FileDownloadCards = (props: FileDownloadCards): JSX.Element => {
             fileName={truncatedFileName(file.name, truncateLength)}
             key={file.name}
             fileExtension={extension(file.name)}
-            actionIcon={<DownloadIconTrampoline />}
+            actionIcon={
+              showSpinner ? (
+                <Spinner size={SpinnerSize.medium} aria-live={'assertive'} />
+              ) : (
+                <ChatCompositeIcon iconName="Download" />
+              )
+            }
             actionHandler={() => {
-              props.downloadHandler && props.downloadHandler(userId, file);
+              if (props.downloadHandler) {
+                setShowSpinner(true);
+                props
+                  .downloadHandler(userId, file)
+                  .then((value: URL | FileDownloadErrorMessage) => {
```

Comment:
`.then` is the old style promise syntax, consider using await syntax:
```ts
setShowSpinner(true);
const url = await props.downloadHandler(userId, file);
if (value instanceof URL) {
  window.open(value.toString(), '_blank', 'noopener,noreferrer');
}
setShowSpinner(false);

Reply:
done